packages/engine/src/scene/systems/ShadowSystem.ts

```diff
@@ -142,18 +143,25 @@ export default async function ShadowSystem(world: World) {
   let dropShadows = new InstancedMesh(shadowGeometry, shadowMaterial, 0)
   dropShadows.matrixAutoUpdate = false
 
-  const dropShadowReactor = startQueryReactor([DropShadowComponent, GroupComponent], function modifyShadowCount() {
+  const dropShadowReactor = startQueryReactor([DropShadowComponent, GroupComponent], function (props) {
+    if (getShadowsEnabled()) return null
+
     world.scene.remove(dropShadows)
     dropShadows = new InstancedMesh(shadowGeometry, shadowMaterial, shadowComponentQuery().length)
     dropShadows.matrixAutoUpdate = false
     dropShadows.layers.disable(ObjectLayers.Camera)
     world.scene.add(dropShadows)
+
+    const sphere = new Sphere()
+    const minRadius = 0.15
+    new Box3().setFromObject(getComponent(props.root.entity, GroupComponent)[0]).getBoundingSphere(sphere)
+    getComponent(props.root.entity, DropShadowComponent).radius = Math.max(sphere.radius * 2, minRadius)
     return null
   })
 
   let sceneObjects = Array.from(Engine.instance.currentWorld.objectLayerList[ObjectLayers.Camera] || [])
 
-  const CreateDropShadows = () => {
+  const UpdateDropShadows = () => {
```

Comment:
let's rename this to `updateDropShadows` to keep variable names consistently camelCase across the codebase

Reply:
Is this the entire system logic? Let's just move it directly into the system execute function